solver/app/app.go

```diff
@@ -108,6 +108,10 @@ func Run(ctx context.Context, cfg Config) error {
 		return errors.Wrap(err, "get contract addresses")
 	}
 
+	if err := approveOutboxes(ctx, network, backends, solverAddr); err != nil {
```

Comment:
This needs to be pulled up or otherwise solver starts solving before all approvals are finished.